Lib/test/test_typing.py

```diff
@@ -4793,61 +4793,110 @@ def test_no_isinstance(self):
 
 class SpecialAttrsTests(BaseTestCase):
     def test_special_attrs(self):
-        cls_to_check = (
+        P = typing.ParamSpec('P')
+        cls_to_check = {
             # ABC classes
-            typing.AbstractSet,
-            typing.AsyncContextManager,
-            typing.AsyncGenerator,
-            typing.AsyncIterable,
-            typing.AsyncIterator,
-            typing.Awaitable,
-            typing.ByteString,
-            typing.Callable,
-            typing.ChainMap,
-            typing.Collection,
-            typing.Container,
-            typing.ContextManager,
-            typing.Coroutine,
-            typing.Counter,
-            typing.DefaultDict,
-            typing.Deque,
-            typing.Dict,
-            typing.FrozenSet,
-            typing.Generator,
-            typing.Hashable,
-            typing.ItemsView,
-            typing.Iterable,
-            typing.Iterator,
-            typing.KeysView,
-            typing.List,
-            typing.Mapping,
-            typing.MappingView,
-            typing.MutableMapping,
-            typing.MutableSequence,
-            typing.MutableSet,
-            typing.OrderedDict,
-            typing.Reversible,
-            typing.Sequence,
-            typing.Set,
-            typing.Sized,
-            typing.Tuple,
-            typing.Type,
-            typing.ValuesView,
+            typing.AbstractSet: 'AbstractSet',
```

Comment:
Not sure if we need to be so strict, checking that it's not an empty string should be enough right?

Then again, it probably won't hurt to have this check in.